### PR TITLE
PE-3582: Add extra step before Snapshot Creation: fetch non-synced custom metadata

### PR DESCRIPTION
**Pull request by matibat**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/38jut0c555lo0
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/77k7cqpmpun00

### CODE REVIEW

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 1]
```diff
@@ -122,9 +195,23 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
     _itemToBeCreated = null;
     _snapshotEntity = null;
     _wasSnapshotDataComputingCanceled = false;
+
+    final drive = await _driveDao.driveById(driveId: driveId).getSingle();
+    final isPrivate = drive.privacy == DrivePrivacy.private;
+
+    if (isPrivate) {
+      // TODO: re-visit
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:203]: BUMP -

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 2]
```diff
@@ -122,9 +195,23 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
...
+        final profileKey = profileState.cipherKey;
+        _maybeDriveKey = await _driveDao.getDriveKey(driveId, profileKey);
+      } else {
+        _maybeDriveKey = await _driveDao.getDriveKeyFromMemory(driveId);
+      }
+    }
   }
 
-  void _setTrustedRange(Range? range) {
+  Future<void> _setTrustedRange(Range? range) async {
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:214]: Unnecessary change?

File: lib/components/create_snapshot_dialog.dart [hunk 3]
```diff
@@ -193,7 +196,7 @@ String _loadingDialogDescription(
       return appLocalizationsOf(context).thisMayTakeAWhile;
     }
   } else {
-    // Description for uploading snapshot
+    // Description for preparing and uploading snapshot
     return '';
```
Comment (matibat) [inline lib/components/create_snapshot_dialog.dart:200]: Do we want to have special wording for the new phase? Maybe only the title is enough

File: lib/entities/drive_entity.dart [hunk 4]
```diff
@@ -60,7 +60,8 @@ class DriveEntity extends Entity {
         ..txId = transaction.id
         ..ownerAddress = transaction.owner.address
         ..bundledIn = transaction.bundledIn?.id
-        ..customMetadata = customMetaDataFromData(entityJson)
+        // TODO: Uncomment this
+        /// ..customMetadata = customMetaDataFromData(entityJson)
```
Comment (matibat) [inline lib/entities/drive_entity.dart:64]: Un-do

File: lib/entities/file_entity.dart [hunk 5]
```diff
@@ -85,7 +85,8 @@ class FileEntity extends Entity {
         ..txId = transaction.id
         ..ownerAddress = transaction.owner.address
         ..bundledIn = transaction.bundledIn?.id
-        ..customJsonMetaData = customMetaDataFromData(entityJson)
+        // TODO: Uncomment this
+        /// ..customJsonMetaData = customMetaDataFromData(entityJson)
```
Comment (matibat) [inline lib/entities/file_entity.dart:89]: Un-do

File: lib/entities/folder_entity.dart [hunk 6]
```diff
@@ -57,7 +57,8 @@ class FolderEntity extends Entity {
         ..parentFolderId = transaction.getTag(EntityTag.parentFolderId)
         ..txId = transaction.id
         ..ownerAddress = transaction.owner.address
-        ..customJsonMetaData = customMetaDataFromData(entityJson)
+        // TODO: Uncomment this
+        /// ..customJsonMetaData = customMetaDataFromData(entityJson)
```
Comment (matibat) [inline lib/entities/folder_entity.dart:61]: Un-do

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart [hunk 7]
```diff
@@ -78,8 +81,78 @@ class CreateSnapshotCubit extends Cubit<CreateSnapshotState> {
...
+        customMetaData,
+      );
+    }
+  }
+
+  Future<void> _fetchMissingCustomMetadataForFolder() async {
+    final folderRevisions =
+        await _driveDao.revisionsForFoldersWithNoMetadata(_driveId);
+    logger.d('There are ${folderRevisions.length} folders with no metadata');
+    for (final folderRevision in folderRevisions) {
```
Comment (matibat) [inline lib/blocs/create_snapshot/create_snapshot_cubit.dart:122]: We should do it in batches - same for drive and file

**Review by matibat — COMMENTED** [review]
Self-review checkpoint

**Comment by matibat** [comment]
The tests for CreateSnapshotCubit are failing

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart (hunk 2 above)
Comment (matibat) [reply lib/blocs/create_snapshot/create_snapshot_cubit.dart]: Yes - 98f3930d25a03a2acfe6f288fd042dc951a7f246

File: lib/entities/drive_entity.dart (hunk 4 above)
Comment (matibat) [reply lib/entities/drive_entity.dart]: 93094c32d0266245e682efad7adceefac1e26989

File: lib/blocs/create_snapshot/create_snapshot_cubit.dart (hunk 7 above)
Comment (matibat) [reply lib/blocs/create_snapshot/create_snapshot_cubit.dart]: f17a9a50376504e77edeb018c707a8fa3bb7e77e

File: lib/services/arweave/arweave_service.dart [hunk 8]
```diff
@@ -983,6 +983,10 @@ class ArweaveService {
   ) async {
     // TODO: PE-2917
 
+    if (driveKey != null) {
+      throw UnimplementedError('Drive key decryption not implemented yet.');
+    }
```
Comment (matibat) [inline lib/services/arweave/arweave_service.dart:988]: The driveKey parameter on this method wasn't being used anywhere in the code. It started being used here ofr snapshot creation.

**Review by matibat — COMMENTED** [review]
Self-review checkpoint

**Review by thiagocarvalhodev — APPROVED** [review]
